chcemvediet/apps/anonymization/tests/test_management_commands.py

```diff
@@ -0,0 +1,112 @@
+
+from chcemvediet.apps.anonymization.models import AttachmentFinalization
+from chcemvediet.tests import ChcemvedietTestCaseMixin
+
+
+class AttachmentAnonymizationManagementCommandTest(ChcemvedietTestCaseMixin, TestCase):
+
+    def _pre_setup(self):
+        super(AttachmentAnonymizationManagementCommandTest, self)._pre_setup()
+        self.tempdir = TempDirectory()
+        self.tempdir.write(u'testfile.txt', u'Default testing content')
+        self.filename = os.path.join(self.tempdir.path, u'testfile.txt')
+        self.attachment = self._create_attachment()
+
+    def _post_teardown(self):
+        self.tempdir.cleanup()
+        super(AttachmentAnonymizationManagementCommandTest, self)._post_teardown()
+
+    def _create_attachment(self, **kwargs):
+        return super(AttachmentAnonymizationManagementCommandTest, self)._create_attachment(
+                generic_object=self.action,
+                **kwargs
+                )
+
+
+    def test_attachment_and_file_arguments(self):
+        call_command(u'attachment_anonymization', self.attachment.pk, self.filename)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.attachment.pk, self.attachment.pk)
+        self.assertEqual(attachment_finalization.file.read(), u'Default testing content')
+        self.assertEqual(attachment_finalization.successful, True)
+
+    def test_attachment_argument_may_not_be_omitted(self):
+        with self.assertRaisesMessage(CommandError, u'attachment_anonymization takes at least 1 argument (0 given).'):
+            call_command(u'attachment_anonymization')
+
+    def test_non_existent_attachment_raises_exception(self):
+        with self.assertRaisesMessage(CommandError, u'Attachment instance with pk "-1" does not exist.'):
+            call_command(u'attachment_anonymization', u'-1', self.filename)
+
+    def test_invalid_attachment_id_raises_exception(self):
+        with self.assertRaisesMessage(CommandError, u'Attachment instance with pk "invalid_id" does not exist.'):
+            call_command(u'attachment_anonymization', u'invalid_id', self.filename)
+
+    def test_command_with_too_many_arguments(self):
+        with self.assertRaisesMessage(CommandError, u'attachment_anonymization takes at most 2 arguments (3 given).'):
+            call_command(u'attachment_anonymization', self.attachment.pk, self.filename, u'filename2')
+
+    def test_content_is_read_from_stdin_if_file_argument_is_omitted(self):
+        self.addCleanup(setattr, sys, u'stdin', sys.stdin)
+        sys.stdin = StringIO(u'Content from stdin.')
+        call_command(u'attachment_anonymization', self.attachment.pk)
+        attachment_finalization = AttachmentFinalization.objects.get(attachment=self.attachment)
+        self.assertEqual(attachment_finalization.file.read(), u'Content from stdin.')
+
+    def test_file_argument_and_stdin_together_may_not_be_omitted(self):
```

Comment:
@martinmacko47 Vsetky testy prejdu, ak ich spustam cez terminal. Ak som ich spustal cez IDE, nastala rovnaka chyba. Problem nastava v `sys.stdin.isatty()`. Da sa to vyriesit (* aspon v IDE, na githube som to neskusal) pomocou [stackowerflow](https://stackoverflow.com/questions/8389155/how-to-convince-python-stream-objects-to-return-true-from-isatty)
```
    class PseudoTTY(object):
        def __init__(self, underlying):
            self.__underlying = underlying

        def __getattr__(self, name):
            return getattr(self.__underlying, name)

        def isatty(self):
            return True
    sys.stdin = PseudoTTY(sys.stdin)
```
Najblizsie sa nato pozriem a skusim najs nejake riesenie.

Reply:
@viliambalaz Toto je po vyhodeni toho `isatty` vyriesene, tak?

Reply:
@martinmacko47  Ano tak. Vlastne `class PseudoTTY` som ani nepridaval do kodu, iba som si to vyskusal lokalne. Do testu treba pridat `sys.stdin = StringIO(u'')`, inak by sa test zastavil a cakal by na vstup.